src/services/agents/coach_agent.py

```
@@ -0,0 +1,228 @@
+    )
+
+
+def diagnose_user_state(state: CoachAgentState) -> dict:
+    """
+    Takes the user's natural language request and extracts structured information
+    about their training preferences using an LLM. This forms the first node
+    in our graph.
+    """
+    print("---NODE: Diagnosing User State---")
+    if not state["messages"]:
+        raise ValueError("No messages found in state.")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)